agent/dockerclient/dockerauth/ecr.go

```diff
@@ -91,6 +95,7 @@ func (authProvider *ecrAuthProvider) GetAuthconfig(image string,
 	// containers pull with the same role can be cached
 	if authData.GetPullCredentials() != (credentials.IAMRoleCredentials{}) {
 		key.roleARN = authData.GetPullCredentials().RoleArn
+		key.credentialsID = authData.GetPullCredentials().CredentialsID
```

Comment:
This will likely lead to more getAuthorzationToken calls from customer accounts as I think there will be no cache hits. Throttling limit for this API is "Each supported Region: 500 per second". Should we be worried about this?

Reply:
I think the biggest risk here would be for customers who run very large instances with lots of tasks on a single instance (and those tasks are also using a task execution role). Currently this customer might be seeing a high rate of cache hits if all of their tasks reuse the same task execution role.

On the other hand, hopefully a customer such as this already has some mitigations in place to not need as many image pulls. Also, 500 tps is a very high limit so I wouldn't think this would come up often.

I think that if the ECS backend is telling us that the credentials are unique (via a unique credentials ID), then the right thing to do is for the agent to refresh creds.

Reply:
That makes sense to me. Thanks.